actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,487 @@
+                        *zip(*list(dataset.iterrows())),
+                        [diagnostic_sheet_dir for i in range(len(dataset))],
+                        [overwrite for i in range(len(dataset))],
+                        [this_metadata for i in range(len(dataset))],
+                        batch_size=batch_size,
+                    )
+                # Generate diagnostic sheet dataset rows
+                for r in diagnostic_sheet_result:
+                    if isinstance(r, DiagnosticSheetResult):
+                        diagnostic_sheet_result_dataset.append(
+                            {
+                                DatasetFields.CellId: r.cell_id,
+                                DatasetFields.DiagnosticSheetPath: r.save_path,
+                            }
+                        )
+                    else:
+                        errors.append(
+                            {DatasetFields.CellId: r.cell_id, "Error": r.error}
+                        )
+
+                # Convert diagnostic sheet paths rows to dataframe
+                diagnostic_sheet_result_dataset = pd.DataFrame(
+                    diagnostic_sheet_result_dataset
+                )
+
+                # Drop the various diagnostic sheet columns if they already exist
+                # Check at j = 0 because the path will exist at j > 1 if
+                # multiple metadata
+                drop_columns = []
+                if DatasetFields.DiagnosticSheetPath in dataset.columns and j == 0:
+                    drop_columns.append(DatasetFields.DiagnosticSheetPath)
+
+                dataset = dataset.drop(columns=drop_columns)
+
+                # Update manifest with these paths if there is data
+                if len(diagnostic_sheet_result_dataset) > 0:
+
+                    # Join original dataset to the fov paths
+                    dataset = dataset.merge(
+                        diagnostic_sheet_result_dataset, on=DatasetFields.CellId
+                    )
+
+                    # If j > 0 (i.e. multiple metadata), we will append new paths
+                    # to the same DiagnosticSheetPath column
+                    if j > 0:
+                        path_list = dataset[
+                            [
+                                DatasetFields.DiagnosticSheetPath + "_x",
+                                DatasetFields.DiagnosticSheetPath + "_y",
+                            ]
+                        ].values.tolist()
+
+                        # Flatten pathlist
+                        path_list = [
+                            list(flatten(this_path)) for this_path in path_list
+                        ]
+                        dataset[DatasetFields.DiagnosticSheetPath] = path_list
+
+                        # Delete the _x and _y columns that are auto generated
+                        del dataset[DatasetFields.DiagnosticSheetPath + "_x"]
+                        del dataset[DatasetFields.DiagnosticSheetPath + "_y"]
```

Comment:
Instead of merging with the input dataset, just make a new one of primarily the DiagnosticSheetPaths